classes/cave.js

```diff
@@ -542,10 +443,98 @@ class Cave {
 
             if (reaction.emoji.name === this.adminEmojis.first().name) {
                 await this.newRole(adminConsole, promptUserId);
+
+                // let admin know the action was succesfull
+                adminConsole.send('<@' + promptUserId + '> The role has been added!').then(msg => msg.delete({ timeout: 8000 }));
+            } else if (reaction.emoji.name === Array.from(this.adminEmojis.keys())[1]) {
+                let all = await Prompt.yesNoPrompt('Type "yes" if you would like to delete all mentor channels, "no" if you would like to only delete some.', adminConsole, promptUserId);
+                if (all) {
+                    let deleteNow = await Prompt.yesNoPrompt('All ticket categories older than a minute will be deleted. Type "yes" to confirm or "no" to set a different timeframe. Careful - this cannot be undone!', adminConsole, promptUserId);
+                    var age;
+                    adminConsole.guild.channels.cache.forEach(async channel => {
+                        (deleteNow) ? age = 1 : age = await Prompt.numberPrompt('Enter the number of minutes. All ticket channels older than this time will be deleted. Careful - this cannot be undone!', adminConsole, promptUserId);
+                        if (channel.type === 'category' && channel.name.startsWith('Ticket -') && (Date.now() - channel.createdTimestamp > age * 60)) {
+                            await channel.children.forEach(async child => await discordServices.deleteChannel(child));
+                            await discordServices.deleteChannel(channel);
+                        }
+                    });
+                } else {
+                    var exclude = await Prompt.yesNoPrompt('Type "yes" if you would like to delete all ticket channels **except** for the ones you mention, "no" if you would like for the channels you mention to be deleted.', adminConsole, promptUserId);
+                    var prompt;
+                    if (exclude) {
+                        prompt = 'In **one** message, send all the ticket numbers to be excluded, separated by spaces.\n' +
+                            'To save a specific channel, mention that channel in the same message. Careful - this cannot be undone!';
+                    } else {
+                        prompt = 'In **one** message, send all the ticket numbers to be deleted, separated by spaces.\n' +
+                            'To delete a specific channel, mention that channel in the same message. Careful - this cannot be undone!';
+                    }
+                    var response = await Prompt.messagePrompt(prompt, 'string', adminConsole, admin.id, 30);
+                    var categoryMentions = [];
+                    if (response != null) {
```

Comment:
For this part of the code, if we have a list of all the tickets, and the tickets have the category channel, could we just loop over the tickets, and check the category channel creation time. Also, if the tickets are in a map or collection, and the key is the ticket number, then you can filter or remove those tickets from a copy list when a user wants to not delete some tickets.

What do you think?